cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -184,158 +173,220 @@ public void launchTask(final ExecutorDriver driver, final TaskInfo task) {
             final SlaveStatusDetails details = SlaveStatusDetails.newBuilder()
                 .setStatusDetailsType(SlaveStatusDetails.StatusDetailsType.ERROR_DETAILS)
                 .setSlaveErrorDetails(
-                    SlaveErrorDetails.newBuilder()
-                        .setMsg(msg)
-                        .setDetails(e.getMessage())
+                        SlaveErrorDetails.newBuilder()
+                                .setMsg(msg)
+                                .setDetails(e.getMessage() != null ? e.getMessage() : "-")
                 )
-                .build();
+                    .build();
             final TaskStatus taskStatus = taskStatus(task, TaskState.TASK_ERROR, details);
             driver.sendStatusUpdate(taskStatus);
         }
         LOGGER.debug(taskIdMarker, "< launchTask(driver : {}, task : {})", driver, protoToString(task));
     }
 
+    private void startJob(ExecutorDriver driver, TaskInfo task, TaskDetails taskDetails) {
```

Comment:
All return points of `startJob` should send some sort of taskStatus. Please change this method to return `TaskStatus` which can then be sent by the driver, rather than having many points where `sendStatusUpdate` can be called.